applicationset/generators/generator_spec_processor.go

```diff
@@ -71,8 +71,8 @@ func Transform(requestedGenerator argoprojiov1alpha1.ApplicationSetGenerator, al
 		}
 		var filterParams []map[string]interface{}
 		for _, param := range params {
-
-			if requestedGenerator.Selector != nil && !selector.Matches(labels.Set(keepOnlyStringValues(param))) {
+			flatParam := flattenParameters(param)
```

Comment:
We're using https://github.com/jeremywohl/flatten elsewhere. Is it not a good fit for this use case?

Reply:
I’ll look in to this.

Reply:
So would this be acceptable?

```go
flatParam, err := flatten.Flatten(param, "", flatten.DotStyle)
if err != nil {
	log.WithError(err).WithField("generator", g).
		Error("error flattening params")
	if firstError == nil {
		firstError = err
	}
	continue
}

strParam := make(map[string]string, len(flatParam))
for k, v := range flatParam {
	strParam[k] = fmt.Sprintf("%v", v)
}

if requestedGenerator.Selector != nil && !selector.Matches(labels.Set(strParam)) {
	continue
}
filterParams = append(filterParams, param)
```

The Sprintf call to convert the interface value into a string feels a little janky but I guess most non-scalar values are handled by the flatten call and there's no real way for a user to put a complex type in there anyway.

Reply:
Agreed, that has a janky feel to it. Maybe wrap the conversion and also use it in the other place we're calling `flatten.Flatten`? They both have to convert to string.

Reply:
Interestingly, the git generator actually uses the same sprintf conversion which is the only other occurrence of the Flatten function being used. I'll extract that out and update both references.

Reply:
Updated and removed the custom flatten implementation in place of the flatten library. I've wrapped it in a function to handle to type conversion from `map[string]interface{}` to `map[string]string` to clean things up.
However, the git generator wants a `map[string]interface{}` so no need for the conversion here so I have left that as-is.